runtime/shared_common/CacheMap.cpp

```diff
@@ -3016,43 +3062,46 @@ SH_CacheMap::findCompiledMethod(J9VMThread* currentThread, const J9ROMMethod* ro
 	}
 
 	result = (const U_8*)findROMClassResource(currentThread, romMethod, localCMM, &descriptor, true, NULL, flags);
+	if (NULL != result) {
 #if !defined(J9ZOS390) && !defined(AIXPPC)
-	if (_metadataReleased
+		if (_metadataReleased
 #if defined(LINUX)
-			&& J9_ARE_ALL_BITS_SET(*_runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE)
+		&& J9_ARE_ALL_BITS_SET(*_runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE)
 #endif
-	) {
-		if (TrcEnabled_Trc_SHR_CM_findCompiledMethod_metadataAccess
-		                   && ((uintptr_t) result >= _minimumAccessedShrCacheMetadata)
-		                   && ((uintptr_t) result <= _maximumAccessedShrCacheMetadata)
 		) {
-			J9InternalVMFunctions* vmFunctions = currentThread->javaVM->internalVMFunctions;
-			J9ClassLoader* loader = NULL;
-			J9ROMClass* romClass = vmFunctions->findROMClassFromPC(currentThread, (UDATA)romMethod, &loader);
-			if (NULL != romClass) {
-				J9UTF8* romClassName = J9ROMCLASS_CLASSNAME(romClass);
-				J9UTF8* romMethodName = J9ROMMETHOD_NAME(romMethod);
-				J9UTF8* romMethodSig = J9ROMMETHOD_SIGNATURE(romMethod);
-				Trc_SHR_CM_findCompiledMethod_metadataAccess(
-						currentThread,
-						J9UTF8_LENGTH(romClassName),
-						J9UTF8_DATA(romClassName),
-						J9UTF8_LENGTH(romMethodName),
-						J9UTF8_DATA(romMethodName),
-						J9UTF8_LENGTH(romMethodSig),
-						J9UTF8_DATA(romMethodSig),
-						result
-				);
+			if (TrcEnabled_Trc_SHR_CM_findCompiledMethod_metadataAccess
+			&& ((uintptr_t) result >= _minimumAccessedShrCacheMetadata)
+			&& ((uintptr_t) result <= _maximumAccessedShrCacheMetadata)
+			) {
+				J9InternalVMFunctions* vmFunctions = currentThread->javaVM->internalVMFunctions;
+				J9ClassLoader* loader = NULL;
+				J9ROMClass* romClass = vmFunctions->findROMClassFromPC(currentThread, (UDATA)romMethod, &loader);
+				if (NULL != romClass) {
+					J9UTF8* romClassName = J9ROMCLASS_CLASSNAME(romClass);
+					J9UTF8* romMethodName = J9ROMMETHOD_NAME(romMethod);
+					J9UTF8* romMethodSig = J9ROMMETHOD_SIGNATURE(romMethod);
+					Trc_SHR_CM_findCompiledMethod_metadataAccess(
+							currentThread,
+							J9UTF8_LENGTH(romClassName),
+							J9UTF8_DATA(romClassName),
+							J9UTF8_LENGTH(romMethodName),
+							J9UTF8_DATA(romMethodName),
+							J9UTF8_LENGTH(romMethodSig),
+							J9UTF8_DATA(romMethodSig),
+							result
+							);
+				}
 			}
-		}
-	} else
+		} else
 #if defined(LINUX)
 		if (J9_ARE_ALL_BITS_SET(*_runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE))
 #endif
 		{
 			updateAccessedShrCacheMetadataBounds(currentThread, (uintptr_t *) result);
```

Comment:
Please fix the bug and check result is not NULL before calling updateAccessedShrCacheMetadataBounds().

Reply:
Sorry, missed the null check added above.